ast/src/main/java/com/graphicsfuzz/common/util/UpgradeShadingLanguageVersion.java

```diff
@@ -92,6 +173,73 @@ public void visitVariableIdentifierExpr(VariableIdentifierExpr variableIdentifie
     }
   }
 
+  @Override
+  public void visitVariablesDeclaration(VariablesDeclaration variablesDeclaration) {
+    final Type baseType = variablesDeclaration.getBaseType();
+    if (baseType.hasQualifier(TypeQualifier.VARYING)) {
+      ((QualifiedType) baseType).replaceQualifier(TypeQualifier.VARYING,
+          TypeQualifier.SHADER_INPUT);
+    }
+    visit(baseType);
```

Comment:
Can you add a brief comment explaining the overall purpose of this loop?

Reply:
Done